packages/contracts/contracts/factory/DAOFactory.sol

```diff
@@ -2,155 +2,152 @@
-        VoteConfig calldata _voteConfig,
-        TokenFactory.TokenConfig calldata _tokenConfig,
-        TokenFactory.MintConfig calldata _mintConfig,
-        address _trustedForwarder
-    )
+    function createDao(DAOSettings calldata _daoSettings, PluginSettings[] calldata pluginSettings)
         external
-        returns (
-            DAO dao,
-            ERC20Voting voting,
-            ERC20VotesUpgradeable token,
-            MerkleMinter minter
-        )
+        returns (DAO dao)
     {
-        if (_mintConfig.receivers.length != _mintConfig.amounts.length)
-            revert MintArrayLengthMismatch({
-                receiversArrayLength: _mintConfig.receivers.length,
-                amountsArrayLength: _mintConfig.amounts.length
-            });
-
-        dao = createDAO(_daoConfig, _trustedForwarder);
-
-        // Create token and merkle minter
-        dao.grant(address(dao), address(tokenFactory), dao.ROOT_PERMISSION_ID());
-        (token, minter) = tokenFactory.createToken(dao, _tokenConfig, _mintConfig);
-        dao.revoke(address(dao), address(tokenFactory), dao.ROOT_PERMISSION_ID());
+        // Check if no plugin is provided.
+        if (pluginSettings.length == 0) {
+            revert NoPluginProvided();
+        }
 
-        daoRegistry.register(dao, msg.sender, _daoConfig.name);
+        // Create DAO.
+        dao = _createDAO(_daoSettings);
 
-        voting = createERC20Voting(dao, token, _voteConfig);
+        // Register DAO.
+        daoRegistry.register(dao, msg.sender, _daoSettings.name);
 
-        setDAOPermissions(dao, address(voting));
-
-        emit DAOCreated(_daoConfig.name, address(token), address(voting));
-    }
+        // Grant the temporary permissions.
+        // Grant Temporarly `ROOT_PERMISSION_ID` to `pluginSetupProcessor`.
+        dao.grant(address(dao), address(pluginSetupProcessor), dao.ROOT_PERMISSION_ID());
 
-    /// @notice Creates a new DAO with the `AllowlistVoting` component installed.
-    /// @param _daoConfig The name and metadata hash of the DAO.
-    /// @param _voteConfig The configuration used to set up the the majority voting.
-    /// @param _allowlistVoters An array of addresses that are allowed to vote.
-    /// @param _trustedForwarder The address of the trusted forwarder required for meta transactions.
-    function createAllowlistVotingDAO(
-        DAOConfig calldata _daoConfig,
-        VoteConfig calldata _voteConfig,
-        address[] calldata _allowlistVoters,
-        address _trustedForwarder
-    ) external returns (DAO dao, AllowlistVoting voting) {
-        dao = createDAO(_daoConfig, _trustedForwarder);
-
-        daoRegistry.register(dao, msg.sender, _daoConfig.name);
-
-        voting = createAllowlistVoting(dao, _allowlistVoters, _voteConfig);
-
-        setDAOPermissions(dao, address(voting));
+        // Grant Temporarly `PROCESS_INSTALL_PERMISSION_ID` on `pluginSetupProcessor` to this `DAOFactory`.
+        dao.grant(
```

Comment:
this seems wrong. `dao`(user's created dao) that you are granting permission on is totally different from the dao pluginsetupprocessor responds to. 

pluginsetupprocessor will have the dao(aragon dao) that it will respond too. in my opinion, that line shouldn't work  !

how did it succeed in tests ?

Reply:
pplyXYZ uses modifier canApply(_dao, permissionId)
this _dao is the user dao not managingDao,
and looking at https://github.com/aragon/core/blob/develop/packages/contracts/contracts/plugin/PluginSetupProcessor.sol#L584
 the modifier says actions are allowed if one of the following conditions are met:
a- msg.sender is the _dao (user dao) so any dao can install plugin on itself calling pluginSetupProcessor
b- msg.sender is permitted to do this call by the _dao (user dao)
That is why the tests on daoFactory can pass ..etc